control-plane-operator/controllers/hostedcontrolplane/olm/catalogs.go

```diff
@@ -86,36 +87,19 @@ func ReconcileRedHatOperatorsDeployment(deployment *appsv1.Deployment, ownerRef
 
 func reconcileCatalogDeployment(deployment *appsv1.Deployment, ownerRef config.OwnerRef, dc config.DeploymentConfig, sourceDeployment *appsv1.Deployment) error {
 	ownerRef.ApplyTo(deployment)
+	deployment.Annotations = sourceDeployment.Annotations
```

Comment:
You should preserve other annotations not set in the template. All tests are failing with:

```
WARNING: Object got updated more than one time without a no-op update, this indicates hypershift incorrectly reverting defaulted values
```

```
  &v1.Deployment{
        TypeMeta: {Kind: "Deployment", APIVersion: "apps/v1"},
        ObjectMeta: v1.ObjectMeta{
                ... // 9 identical fields
                DeletionGracePeriodSeconds: nil,
                Labels:                     {"hypershift.openshift.io/managed-by": "control-plane-operator"},
                Annotations: map[string]string{
-                       "deployment.kubernetes.io/revision": "347",
                        "image.openshift.io/triggers":       `[{"from":{"kind":"ImageStreamTag","name":"catalogs:redhat-market`...,
                },
                OwnerReferences: {{APIVersion: "hypershift.openshift.io/v1beta1", Kind: "HostedControlPlane", Name: "example-wq9px", UID: "372fe123-ea6b-47b1-ada8-ad6d874cefcc", ...}},
                Finalizers:      nil,
                ManagedFields:   {{Manager: "hypershift-controlplane-manager", Operation: "Update", APIVersion: "apps/v1", Time: s"2023-01-13 04:43:45 +0000 UTC", ...}, {Manager: "kube-controller-manager", Operation: "Update", APIVersion: "apps/v1", Time: s"2023-01-13 04:43:49 +0000 UTC", ...}},
```